local_planner/src/nodes/local_planner_node.cpp

```diff
@@ -470,15 +461,21 @@ void LocalPlannerNode::publishBox() {
 }
 
 void LocalPlannerNode::publishWaypoints(bool hover) {
-  const ros::Time now = ros::Time::now();
+  // Don't publish waypoints if the drone isn't armed or mission has been
+  // enabled. Need second check to avoid race condition
+  if (!armed_ && !mission_) {
```

Comment:
this change breaks offboard and rtl

Reply:
I'm currently experimenting with this part, since it is the cause of some of the issues I'm still having. But why do you think it will break offboard? In offboard the drone is armed, so this clause won't return...

The main problem is that the smoothing starts at the current drone location and then converges toward the setpoint. (Which is the intended behavior I just added in this PR!) But this process starts as soon as the waypoint generator is run the first time, so by the time we actually arm and start, the smoothed setpoint will have converged to the setpoint, hence we're sending a step again. 

I wanted to avoid sending the `armed_` boolean into the waypoint generator (thus further complicating its state machine), but I don't see a way around it.

Reply:
yes, sorry I messed up reading the conditions. 
You would like to start generating waypoint when the vehicle is in an OA enabled mode. Would it make sense to check for 
`if(!mission_ && !rtl_ && !offboard_) return;` ?

Reply:
ah no wait.. it could happen in both cases
1. I arm on the ground and stay there for a while
2. I switch to mission mode on the ground and stay there for a while
In both cases you don't want the waypoint generator to start running

Reply:
```bash
bool in_oa_mode = mission_ || rtl_ || offboard_;
if ( (armed_ && !in_oa_mode) || (!armed && in_oa_mode) ) return;

Reply:
Yeah, I pretty much did that, but passing this boolean (I named it `airborne`) into the waypoint generator anyway. Let me know what you think